subiquity/common/filesystem/gaps.py

```diff
@@ -165,3 +165,15 @@ def largest_gap_size(device):
     if largest is not None:
         return largest.size
     return 0
+
+
+def gap_after_partition(partition):
+    pg = parts_and_gaps(partition.device)
+    part_idx = pg.index(partition)
+    try:
+        gap = pg[part_idx + 1]
+    except IndexError:
+        return None
+    if isinstance(gap, Gap):
+        return gap
+    return None
```

Comment:
So I don't know if this is the branch to do it, but I don't think this is quite the right logic or -- more importantly -- interface. I know we've talked about this issue of "sliding" batches of partitions around when edits are made. 

In this branch, if you take an empty 10G disk, add two 1G partitions and then edit the first, there is no gap after it and so the max size is 1G. In main, the max size is 9G and that feels more right. So I think the interface here needs to be "get me the movable partitions after this one and then any trailing gap". And then PartitionStretchy can call this API and only look at the gap bit, and the code that handles the edits can adjust the offsets of the movable partitions.

I hesitate to pile more onto this branch but actually it might not be so bad. The hardest part might be coming up with a name for the API...

Reply:
OK, I'm ejecting what is currently called gap_after_partition from this branch and we'll sort it out.  I can live with max_size being wrong until then.